quantmsrescore/ms2pip.py

```
@@ -309,23 +481,420 @@ def choose_best_ms2pip_tolerance(
+    ObservedSpectrum
+
+    Raises
+    ------
+    UnsupportedSpectrumFiletypeError
+        If the file extension is not supported.
+
+    """
+    try:
+        spectra = OpenMSHelper.get_mslevel_spectra(file_name=str(spec_file), ms_level=2)
+    except ValueError:
+        raise exceptions.UnsupportedSpectrumFiletypeError(Path(spec_file).suffixes)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)